packages/specs/src/runtime/babe.ts

```diff
@@ -0,0 +1,102 @@
+import { RuntimeCallsSpec, RuntimeApisModule } from '@delightfuldot/types';
+
+const V1_V2_SHARED: RuntimeCallsSpec = {
+  currentEpoch: {
+    docs: 'Returns information regarding the current epoch.',
+    params: [],
+    type: 'Epoch',
+  },
+
+  currentEpochStart: {
+    docs: 'Returns the slot that started the current epoch.',
+    params: [],
+    type: 'Slot',
+  },
+
+  nextEpoch: {
+    docs: 'Returns information regarding the next epoch (which was already previously announced).',
+    params: [],
+    type: 'Epoch',
+  },
+
+  generateKeyOwnershipProof: {
+    docs: [
+      'Generates a proof of key ownership for the given authority in the',
+      'current epoch. An example usage of this module is coupled with the',
+      'session historical module to prove that a given authority key is',
+      'tied to a given staking identity during a specific session. Proofs',
+      'of key ownership are necessary for submitting equivocation reports.',
+      'NOTE: even though the API takes a `slot` as parameter the current',
+      'implementations ignores this parameter and instead relies on this',
+      'method being called at the correct block height, i.e. any point at',
+      'which the epoch for the given slot is live on-chain. Future',
+      'implementations will instead use indexed data through an offchain',
+      'worker, not requiring older states to be available.',
+    ],
+    params: [
+      {
+        name: 'slot',
+        type: 'Slot',
+      },
+      {
+        name: 'authorityId',
+        type: 'AccountId32',
+      },
+    ],
+    type: 'Option<OpaqueKeyOwnershipProof>',
+  },
+
+  submitReportEquivocationUnsignedExtrinsic: {
+    docs: [
+      'Submits an unsigned extrinsic to report an equivocation. The caller',
+      'must provide the equivocation proof and a key ownership proof',
+      '(should be obtained using `generate_key_ownership_proof`). The',
+      'extrinsic will be unsigned and should only be accepted for local',
+      'authorship (not to be broadcast to the network). This method returns',
+      '`None` when creation of the extrinsic fails, e.g. if equivocation',
+      'reporting is disabled for the given runtime (i.e. this method is',
+      'hardcoded to return `None`). Only useful in an offchain context.',
+    ],
+    params: [
+      {
+        name: 'equivocationProof',
+        type: 'BabeConfiguration',
+      },
+      {
+        name: 'keyOwnerProof',
+        type: 'OpaqueKeyOwnershipProof',
+      },
+    ],
+    type: 'Option<null>',
```

Comment:
`'Option<null>'` will not work when we decode the responded raw hex data from the server. We need to use a codec type for `null` which is [`Null`](https://github.com/CoongCrafts/delightfuldot/blob/9436bbf6e06e2dc8cd6e75011f50927bc43a391b/packages/shape/src/extension/raw.ts#L57). We would then need to defined a TS type for `Null` which is `export type Null  = null` in the primitive type list in `/known/common.ts`
```suggestion
    type: 'Option<Null>',
```